test/src/test/java/hudson/ExtensionListListenerTest.java

```diff
@@ -65,4 +67,15 @@ public void onChange() {
             onChangeCallCount++;
         }
     }
+
+    /**
+     * Check that dynamically loading a plugin containing an extension that is used by an {@link ExtensionListListener}
+     * and that loads other extensions internally does not lead to extension lists with duplicate entries.
+     */
+    @Test
+    public void checkDynamicLoad_singleRegistration() throws Throwable {
+        PluginManagerUtil.dynamicLoad("extension-list-listener-dynamic-load.hpi", r.jenkins);
```

Comment:
Recommend using RJR for this type of test.

Reply:
See [4b33059](https://github.com/jenkinsci/jenkins/pull/10240/commits/4b330592811f04a81373878960e1dcca3563185a).